src/order/upper_lower.lean

```diff
@@ -351,13 +379,277 @@ by simp_rw lower_set.compl_supr
+@[simp] lemma to_dual_Sup (S : set (lower_set α)) :
+  (Sup S).to_dual = Sup (lower_set.to_dual '' S) := set_like.coe_injective (by simp)
+@[simp] lemma of_dual_Inf (S : set (lower_set αᵒᵈ)) :
+  (Inf S).of_dual = Inf (lower_set.of_dual '' S) := set_like.coe_injective (by simpa)
+@[simp] lemma of_dual_Sup (S : set (lower_set αᵒᵈ)) :
+  (Sup S).of_dual = Sup (lower_set.of_dual '' S) := set_like.coe_injective (by simp)
+
+@[simp] lemma to_dual_supr (f : ι → lower_set α) :
+  (⨆ i, f i).to_dual = ⨆ i, (f i).to_dual := set_like.coe_injective (by simp)
+@[simp] lemma to_dual_infi (f : ι → lower_set α) :
+  (⨅ i, f i).to_dual = ⨅ i, (f i).to_dual := set_like.coe_injective (by simpa)
+@[simp] lemma of_dual_supr (f : ι → lower_set αᵒᵈ) :
+  (⨆ i, f i).of_dual = ⨆ i, (f i).of_dual := set_like.coe_injective (by simp)
+@[simp] lemma of_dual_infi (f : ι → lower_set αᵒᵈ) :
+  (⨅ i, f i).of_dual = ⨅ i, (f i).of_dual := set_like.coe_injective (by simpa)
+
+@[simp] lemma to_dual_infi₂ (f : Π i, κ i → lower_set α) :
+  (⨅ i j, f i j).to_dual = (⨅ i j, (f i j).to_dual) := set_like.coe_injective (by simp)
+@[simp] lemma to_dual_supr₂ (f : Π i, κ i → lower_set α) :
+  (⨆ i j, f i j).to_dual = (⨆ i j, (f i j).to_dual) := set_like.coe_injective (by simp)
+@[simp] lemma of_dual_infi₂ (f : Π i, κ i → lower_set αᵒᵈ) :
+  (⨅ i j, f i j).of_dual = (⨅ i j, (f i j).of_dual) := set_like.coe_injective (by simp)
+@[simp] lemma of_dual_supr₂ (f : Π i, κ i → lower_set αᵒᵈ) :
+  (⨆ i j, f i j).of_dual = (⨆ i j, (f i j).of_dual) := set_like.coe_injective (by simp)
+
+end lower_set
+
+@[simp] lemma upper_set.of_dual_to_dual (s : upper_set αᵒᵈ) : s.of_dual.to_dual = s := by {ext,simp}
+@[simp] lemma upper_set.to_dual_of_dual (s : upper_set α) : s.to_dual.of_dual = s := by {ext,simp}
+@[simp] lemma lower_set.of_dual_to_dual (s : lower_set αᵒᵈ) : s.of_dual.to_dual = s := by {ext,simp}
+@[simp] lemma lower_set.to_dual_of_dual (s : lower_set α) : s.to_dual.of_dual = s := by {ext,simp}
+
+end has_le
+
+/-! #### Images -/
+
+section map
+
+section has_le
+
+variables [has_le α] [has_le β] [order_iso_class F α β]
+
+/-- The image of an `upper_set` under an order isomorphism, as an `upper_set`. -/
+def upper_set.map (s : upper_set α) (φ : F) : upper_set β := ⟨φ '' s, s.upper.map φ⟩
+
+/-- The image of a `lower_set` under an order isomorphism, as a `lower_set`. -/
+def lower_set.map (s : lower_set α) (φ : F) : lower_set β := ⟨φ '' s, s.lower.map φ⟩
+
+@[simp] lemma upper_set.coe_map (s : upper_set α) (φ : F) : (s.map φ : set β) = φ '' s := rfl
+@[simp] lemma lower_set.coe_map (s : lower_set α) (φ : F) : (s.map φ : set β) = φ '' s := rfl
+
 end has_le
 
+section preorder
+
+variables [preorder α] [preorder β] [order_hom_class F α β]
+
+/-- The preimage of an `upper_set` under an order homomorphism, as an `upper_set` -/
+def upper_set.comap (t : upper_set β) (φ : F) : upper_set α := ⟨φ ⁻¹' t, t.upper.preimage φ⟩
```

Comment:
Should it be an `order_hom` or this will break dot notation? Anyway, please add `comap_mono`.